specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-09-01-preview/Overview.json

```diff
@@ -0,0 +1,367 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Insights",
+    "description": "API spec for Microsoft.SecurityInsights (Azure Security Insights) resource provider",
+    "version": "2022-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/overview": {
```

Comment:
mm.... Maybe - getOVerview is a more suitable name?
we are not exposing a resource\a list of resources, we are calculating users aggregation based on their resources and data.

the OverviewSectionKindEnum represents the different overview sections, based on different business aspects.
in this phase we are introducing 4.

Reply:
One can call the overview api and get a set of sections,
 by the section kind we can determine the response structure

Reply:
POST Apis are reserved to execution actions, you are using POST to return a list of items which is not recommended (see this: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)),  is better to model the API as a resource type. e.g. `/Microsoft.SecurityInsights/overviews` 

How are `overviews` set? does the customer set these values or is your RP setting them.

What does a customer can do with these values? This API is still unclear to me.

Reply:
I can explain.
Our customers have a lot of information saved in their LogAnalytics workspace (which is a resource our product is based on) and have a lot of configurations, managed by our RP.

Our purpose is to create an API for aggregations over these data in LA WS and configurations.
Hence Overview.

Because our product has lots of business aspects we have a section per business aspect.
the reason this is not a resource is because the response is changing all the time.
for example, we aggregate the amount of data the user has ingested, we aggregate the number of incidents that were created\closed

We are using this API to build some sort of dashboard. Users can use it for that reason too, but it is mostly used by our portal:
![image](https://user-images.githubusercontent.com/58462609/169239437-acd3cb1e-c883-4438-8215-5ded698705f2.png)

I hope this helps understanding,

Moreover, previously to this PR we have met with ARM representatives, this PR is the meeting outcome.
They have encouraged us to expose this API in this manner

Reply:
Final question, is aggregating data a process that needs reaching out to different data stores? Can the process be completed in <= 2secs?

Reply:
We are adding more documentation on when to add an exception and allow POST to retrieve data, if this API is meant to paginate the response, then we'll need to follow a different approach (so please let me know if this is the case), hopefully it is not, but as a general rule we are allowing POST APIs to return data when the RP needs to aggregate data from different data sources and the API takes over than 2 seconds (which is the requirement for sync APIs), and the idea is to convert this API to an async API instead: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously

Reply:
Checked offline, no pagination is needed and the data aggregates info from different data sources. Signing off from ARM side